conu/backend/podman/backend.py

```diff
@@ -0,0 +1,152 @@
+
+    :param reference: str, e.g. (registry.fedoraproject.org/fedora:27)
+    :return: collection (tuple or list), ("registry.fedoraproject.org/fedora", "27")
+    """
+    if ":" in reference:
+        im, tag = reference.rsplit(":", 1)
+        if "/" in tag:
+            # this is case when there is port in the registry URI
+            return (reference, "latest")
+        else:
+            return (im, tag)
+
+    else:
+        return (reference, "latest")
+
+
+class PodmanBackend(Backend):
+    """
+    For more info on using the Backend classes, see documentation of
+    the parent :class:`conu.apidefs.backend.Backend` class.
+    """
+    name = "podman"
+    ContainerClass = PodmanContainer
+    ImageClass = PodmanImage
+
+    def __init__(self, logging_level=logging.INFO, logging_kwargs=None, cleanup=None):
+        """
+        This method serves as a configuration interface for conu.
+
+        :param logging_level: int, control logger verbosity: see logging.{DEBUG,INFO,ERROR}
+        :param logging_kwargs: dict, additional keyword arguments for logger set up, for more info
+                                see docstring of set_logging function
+        :param cleanup: list, list of cleanup policy values, examples:
+            - [CleanupPolicy.EVERYTHING]
+            - [CleanupPolicy.VOLUMES, CleanupPolicy.TMP_DIRS]
+            - [CleanupPolicy.NOTHING]
+        """
+
+        super(PodmanBackend, self).__init__(
+            logging_level=logging_level, logging_kwargs=logging_kwargs, cleanup=cleanup)
+
+    def cleanup_containers(self):
+        # TODO: Test this
+        conu_containers = self._list_podman_containers(filter=CONU_ARTIFACT_TAG)
+        for c in conu_containers:
+            logger.info("Trying to remove conu container: %s" % c)
+            logger.debug("Removing container %s created by conu", c)
+            run_cmd(["podman", "stop", c])
+            run_cmd(["podman", "rm", c])
+
+    def list_containers(self):
+        """
+        List all available podman containers.
+
+        :return: collection of instances of :class:`conu.PodmanContainer`
+        """
+        containers = []
+        for identifier in self._list_podman_containers():
+            inspect_data = PodmanContainer._inspect(identifier)
+            name = graceful_get(inspect_data, "Name")
+            image_id = graceful_get(inspect_data, "ImageID")
+
+            try:
+                image_name, image_tag = parse_reference(inspect_data["ImageName"])
+            except (IndexError, TypeError):
+                image_name, image_tag = None, None
+
+            image = PodmanImage(image_name, tag=image_tag, identifier=image_id)
+            container = PodmanContainer(image, identifier, name=name)
+            inspect_to_container_metadata(container.metadata, inspect_data, image)
+            containers.append(container)
+
+        return containers
+
+    def list_images(self):
```

Comment:
I have ~800 podman images locally and it takes forever to do this. Would it be possible to not do the inspecting?

Reply:
I think it is, to do this `_list_podman_containers()` should return not only container ID, but also images names, so it will be possible to create new image instance with no inspect. I will fix this.

Reply:
This can help:
```
podman images --format json
```

Reply:
With no inspect we cannot fetch container metadata with inspect_to_container_metadata() function, that is if you want to access metadata from container from `list_containers()` you have to run `get_metadata()` method that will run inspection. I'm not sure if it is okay?

Reply:
Okay I misunderstood you, you are talking about list_images, not list_containers.

Reply:
There is a problem with no inspection, that we lose opportunity to get image's/container's metadata. So if you've got your image/container from list_images/list_containers you have to run manually `get_metadata()` method that will run inspection and fetch metadata.

Reply:
You can easily make the metadata object lazy:
```python
def __init__(self):
  self._metadata = None

@property
def metadata(self):
  if self._metadata is None:
    self._metadata = self.get_metadata()
  return self._metadata
```